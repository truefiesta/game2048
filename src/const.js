export const Move = {
    DOWN: `down`,
    UP: `up`,
    LEFT: `left`,
    RIGHT: `right`,
};