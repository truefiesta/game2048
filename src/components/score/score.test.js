import React from "react";
import renderer from "react-test-renderer";
import Score from "./score";
import {ScoreType} from "../../const";

describe(`Score snapshot`, () => {
    it(`renders with Game title`, () => {
        const score = renderer.create(
            <Score
                title={ScoreType.GAME}
                points={0}
                newPoints={0}
            />
        )
        .toJSON();

        expect(score).toMatchSnapshot();
    });

    it(`renders with Best title`, () => {
        const score = renderer.create(
            <Score
                title={ScoreType.BEST}
                points={3408}
            />
        )
        .toJSON();

        expect(score).toMatchSnapshot();
    });
});
