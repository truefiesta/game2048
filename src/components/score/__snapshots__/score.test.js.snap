// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score snapshot renders with Best title 1`] = `
<div
  className="box"
>
  <h3
    className="subsection-title"
  >
    Best
     score: 
  </h3>
  <p
    className="subsection-title game-score"
  >
    3408
  </p>
</div>
`;

exports[`Score snapshot renders with Game title 1`] = `
<div
  className="box"
>
  <h3
    className="subsection-title"
  >
    Game
     score: 
  </h3>
  <p
    className="subsection-title game-score"
  >
    0
  </p>
</div>
`;
